.github/workflows/build-pr.yml

```diff
@@ -15,6 +15,19 @@ jobs:
     runs-on: self-hosted
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install abigen
+        run: |
+          sudo add-apt-repository -y ppa:ethereum/ethereum
+          sudo apt-get update
+          sudo apt-get -y install ethereum
+
+      - name: Install dependencies
+        run: cd contracts && npm install
+      - name: Generate Solidity Contract Bindings
+        run: cd contracts && npx hardhat generate-abi-bindings --output-dir generated
```

Comment:
This may lead to the committed generated files mismatching the solidity files. 

Instead of this step would add a "Does solidity match the generated files" test in https://github.com/obscuronet/obscuro-internal/issues/1424